python/lsst/image_cutout_backend/_image_cutout_backend.py

```diff
@@ -0,0 +1,412 @@
+        read_result = self.read_ref(stencil, ref)
+        if mask_plane is not None:
+            read_result.mask(mask_plane)
+        return self.write_fits(read_result)
+
+    def process_uuid(
+        self,
+        stencil: SkyStencil,
+        uuid: UUID,
+        *,
+        component: Optional[str] = None,
+        mask_plane: Optional[str] = "STENCIL",
+    ) -> ResourcePath:
+        """Extract and write a cutout from a dataset identified by its UUID.
+
+        Parameters
+        ----------
+        stencil : `SkyStencil`
+            Definition of the cutout region, in sky coordinates.
+        uuid : `UUID`
+            Unique ID of the dataset to read from.
+        component : `str`, optional
+            If not `None` (default), read this component instead of the
+            composite dataset.
+        mask : `str`, optional
+            If not `None`, set this mask plane in the extracted cutout showing
+            the approximate stencil region.  Does nothing if the image type
+            does not have a mask plane.  Defaults to ``STENCIL``.
+
+        Returns
+        -------
+        uri : `ResourcePath`
+            Full path to the extracted cutout.
+        """
+        read_result = self.read_uuid(stencil, uuid, component=component)
+        if mask_plane is not None:
+            read_result.mask(mask_plane)
+        return self.write_fits(read_result)
+
+    def process_search(
+        self,
+        stencil: SkyStencil,
+        dataset_type_name: str,
+        data_id: DataId,
+        collections: Iterable[str],
+        *,
+        mask_plane: Optional[str] = "STENCIL",
+    ) -> ResourcePath:
+        """Extract and write a cutout from a dataset identified by its UUID.
+
+        Parameters
+        ----------
+        stencil : `SkyStencil`
+            Definition of the cutout region, in sky coordinates.
+        dataset_type_name : `str`
+            Name of the butler dataset.  Use ``.``-separate terms to read an
+            image-like component.
+        data_id : `dict` or `DataCoordinate`
+            Mapping-of-dimensions identifier for the dataset within its
+            collection.
+        collections : `Iterable` [ `str` ]
+            Collections to search for the dataset, in the order they should be
+            searched.
+        mask : `str`, optional
+            If not `None`, set this mask plane in the extracted cutout showing
+            the approximate stencil region.  Does nothing if the image type
+            does not have a mask plane.  Defaults to ``STENCIL``.
+
+        Returns
+        -------
+        uri : `ResourcePath`
+            Full path to the extracted cutout.
+        """
+        read_result = self.read_search(stencil, dataset_type_name, data_id, collections)
+        if mask_plane is not None:
+            read_result.mask(mask_plane)
+        return self.write_fits(read_result)
+
+    def read_ref(self, stencil: SkyStencil, ref: DatasetRef) -> ReadResult:
```

Comment:
I don't like the use of "read" here because to me it implies that the cutout has already been calculated and stored and we're just reading the existing file from storage, and my understanding is that this is creating the cutout. Either that, or if this is reading an existing cutout from storage, I have entirely misunderstood how all of this is supposed to work and may need more orientation.

Reply:
How about "extract" or "extract_from" instead of "read"?  Your understanding of what it is doing is correct; I said "read" because I'm used to thinking of the read operation as what actually creates the cutout, because I'm used to thinking of cutouts as in-memory things, not persisted things; between this and the comment on the butler `__init__` parameter docs, I think the lesson here is that I need to be more consistent about "cutout" meaning or implying the persisted thing only, since that's the real deliverable.

Reply:
extract sounds great. That makes sense to me given my mental model of what the Butler is doing under the hood. Thanks!